src/wp-includes/block-template-utils.php

```diff
@@ -17,24 +17,29 @@
  * @return WP_Block_Template|WP_Error Template.
  */
 function _build_template_result_from_post( $post ) {
-	$terms = get_the_terms( $post, 'wp_theme' );
+	$template_type = $post->post_type;
 
-	if ( is_wp_error( $terms ) ) {
-		return $terms;
+	if ( 'wp_template' !== $template_type ) {
+		return new WP_Error( 'template_wrong_post_type', __( 'An invalid post was provided for this template.', 'gutenberg' ) );
```

Comment:
We should remove the "Gutenberg" domain from the translations.

Reply:
We should write a static check and integrate it with CI to point these out 🤔